File: 2021/May/Day 17/countingBits.py
# Source :https://leetcode.com/problems/counting-bits/submissions/
class Solution:
    def countBits(self, num: int) -> List[int]:
        return [self.return_ones(i) for i in range(num+1)];



    def return_ones(self,num):
        count =0;
        while(num):
            if(int(num%2==1)):
                count+=1;
            num = int(num/2)
        return count

File: 2022/July/Day 10/countingBits.cpp
// Source : https://leetcode.com/problems/counting-bits/submissions/
class Solution {
public:

    int retBin(int n)
    {
        int count=0;
        while(n)
        {
            count=n%2==1?count+1:count;
            n/=2;
        }
        return count;

    }

    vector<int> countBits(int n) {

        vector<int> out;
        for(int i=0;i<=n;i++)
        {
            out.insert(out.end(),retBin(i));
        }
        return out;

    }
};
